packages/wallet-sdk/src/sign/util.ts

```diff
@@ -28,21 +28,17 @@ export async function fetchSignerType(params: {
     event: 'selectSignerType',
     data: params.preference,
   };
-  const { data } = await communicator.postMessage(request);
+  const { data } = await communicator.postRequestAndWaitForResponse(request);
   return data as SignerType;
 }
 
 async function listenForWalletLinkSessionRequest(
   communicator: Communicator,
   metadata: AppMetadata
 ) {
-  try {
-    await communicator.onMessage<ConfigMessage>(
-      ({ event }) => event === 'WalletLinkSessionRequest'
-    );
-  } catch (_) {
-    return;
-  }
+  await communicator
+    .onMessage<ConfigMessage>(({ event }) => event === 'WalletLinkSessionRequest')
+    .catch(() => {});
```

Comment:
`WalletLinkSessionRequest` listener can get cancelled before receiving the message.